src/Generation/generator.py

```diff
@@ -0,0 +1,82 @@
+import json
+import regex
+from llm.connector import connector
+
+DESCRIPTION_DIRECTORY = "Generation/Descriptions"
+FORM_FILE_NAME = "form.json"
+QUERY_FILE_NAME = "query_text"
+EXAMPLES_FILE_NAME = "examples.json"
+PREDEFINED_FILE_NAME = "predefined.json"
+FORM_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + FORM_FILE_NAME
+QUERY_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + QUERY_FILE_NAME
+EXAMPLES_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + EXAMPLES_FILE_NAME
+PREDEFINED_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + PREDEFINED_FILE_NAME
+EXAMPLES_KEY = "Examples"
+
+class Form:
+    def __init__(self, form_text: str):
+        self._form = form_text
+        self._keys = list(json.loads(self._form).keys())
+
+    # TODO: Maybe we should apply a more detailed check just to be sure
+    def FormatCheck(self, data_to_check: dict) -> bool:
+        for key in self._keys:
+            if key not in data_to_check:
+                return False
+        return True
+
+    def __str__(self):
+        return self._form
+
+class Generator:
+    __json_pattern = regex.compile(r'\{(?:[^{}]|(?R))*\}')
+
+    def __init__(self, type_name: str):
+        self._type_name = type_name
+        self._form = Form(open(FORM_FILE.format(type_name=type_name), 'r').read())
+        self._template_query_text = open(QUERY_FILE.format(type_name=type_name), 'r').read()
+        examples = json.load(open(EXAMPLES_FILE.format(type_name=type_name), 'r'))
+        self._examples = []
+        for example in examples[EXAMPLES_KEY]:
+            self._examples.append(json.dumps(example, indent=4))
+
+    def generate(self, **kwargs) -> (bool, dict):
+        query_text = self._template_query_text.replace("{{form}}", str(self._form))
+        for (key, value) in kwargs.items():
+            query_text = query_text.replace(f"{{{{{key}}}}}", value)
+        
+        # 0 - Generate without influence
+        # 1 - Generate with a single example
+        # 2 - Generate with multiple examples
+        for i in range(3):
```

Comment:
If I understood this correctly, we first try zero-shot generation (no examples), then add more and more examples until it (hopefully) works. This is risky, because adding the examples might also improve *quality* of the results, not just their correctness.

Then again, I haven't actually tested this.

Reply:
I didn't want to give them any influence in the first round. Let them generate whatever comes to their "mind", maybe it is more creative in that way. Since we are using quite good and large models they can't be so wrong, I think.

Anyway, I tested it with race "Orc".  I wouldn't say GPT generates better. Cohere, on the other hand, doesn't even hit the format without an example. This will not be accepted due to the check and it will then generate an acceptable one with an example. But yes, probably it would be a good idea to simple include all the examples (or at least one) on the first try.

GPT:

without example
{
	"Common": ["Slasher", "Crasher", "Bruiser"],
	"Rare": ["Ravager", "Pummeler", "Destroyer"],
	"Epic": ["Tyrant", "Berserker", "Conqueror"],
	"Legendary": ["Gorgrath the Bloodmoon", "Uzgrom the Destroyer", "Grommuk the Conqueror"]
}

with a single example:
{
	"Common": ["Orc Grunt", "Marauder", "Thug"],
	"Rare": ["Ogre Enforcer", "Vandal", "Assassin"],
	"Epic": ["Berserker Shaman", "Warlord Brutus", "Ravager Witch"],
	"Legendary": ["Grommash the Devastator", "The Blackhand Warlord", "Kargath the Bloodthirsty"]
}

with multiple examples:
{
	"Common": ["Runt", "Pig Face", "Shredder"],
	"Rare": ["Orc Raider", "Warchief", "Shadowblade"],
	"Epic": ["Orcish Warlock", "Savage Berserker", "Spirit Caller"],
	"Legendary": ["Grommash, the Bloodthirsty", "Throk, the Unstoppable", "Ulfang, the Shadow King"]
}

---------------------------------------------------------------
COHERE:

without example:
{
    "Common": [
        "Grunt",
        "Thug",
        "Brute"
    ],
    "Rare": [
        "Warrior",
        "Veteran",
        "Boss"
    ],
    "Epic": [
        "Crushoor",
        "Corrupter",
        "Warcrator"
    ],
    "Legendary": "Kaiser Krush"
}

with a single example:
{
    "Common": ["Brute Orc", "Stouthelm", "Gobspit"],
    "Rare": ["Warrior Chief", "Bloodfist", "Skullcrusher"],
    "Epic": ["Warchief", "Skullraiser", "Carnage Incarnate"],
    "Legendary": ["The Mighty Orbok", "Ghaz'kull, Beast of Horrors", "Zogstuf, Deathknell of Legends"]
}

with multiple examples:
{
    "Common": ["Brute", "Berserker", "Thug"],
    "Rare": ["Warrior", "Arbiter", "Spellbreaker"],
    "Epic": ["Warchief", "Shouter", "Magther"],
    "Legendary": ["Gollumak the Conqueror", "Jughrokhir the Spell-Crusher", "Orbulg the Despoiler"]
}

Reply:
Sooo after some testing with cohere, I added it to use examples by default